# Unit 3 Challenges

##      Challenge 3:1:5 

In [18]:
import pandas as pd
import numpy as np
import statistics

Greg was 14, Marcia was 12, Peter was 11, Jan was 10, Bobby was 8, and Cindy was 6 when they started playing the
Brady kids on The Brady Bunch. Cousin Oliver was 8 years old when he joined the show. 
What are the mean, median,and mode of the kids ages when they first appeared on the show?
What are the variance, standard deviation, and standard error?

In [41]:
df = pd.DataFrame()
ef = pd.DataFrame()

total = 14 + 12 + 11 + 10 + 8 + 8 + 6
total

mean = total / 7

mean = 9.86

median = 10

mode = 8 

((14 - mean) ** 2) / (7 - 1)
((12 - mean) ** 2) / (7 - 1)
((11 - mean) ** 2) / (7 - 1)
((10 - mean) ** 2) / (7 - 1)
((8 - mean) ** 2) / (7 - 1)
((6 - mean) ** 2) / (7 - 1)

(((14 - mean) ** 2) / (7 - 1)) ** (1/2)
(((12 - mean) ** 2) / (7 - 1)) ** (1/2)
(((11 - mean) ** 2) / (7 - 1)) ** (1/2)
(((10 - mean) ** 2) / (7 - 1)) ** (1/2)
(((8 - mean) ** 2) / (7 - 1)) ** (1/2)
(((6 - mean) ** 2) / (7 - 1)) ** (1/2)


In [10]:
df['ages'] = [14, 12, 11, 10, 8, 8, 6 ]

In [15]:
print(np.mean(df['ages']))
ages = df['ages']

9.857142857142858


In [16]:
np.median(ages)

10.0

In [19]:
statistics.mode(ages)

8

In [20]:
ages.var()

7.476190476190475

In [21]:
(ages.var()) ** .5

2.734262327610589

In [22]:
np.std(ages, ddof=1)

2.734262327610589

In [28]:
2.734262327610589/(7 ** 0.5)

1.0334540197243192

2) In this example, I would choose the median as my central tendency, because it removes the largest outliers.  Second, I would choose standard deviation, as my  Variance. With the median it accounts for the other answers (max 12.7 - min 7.27)

3) After, Cindy's birthday the mean, variance, std, and se would all change because they involve the mean. While the median and the mode wouldnt be affected. 

In [24]:
df['ages'] = [14, 12, 11, 10, 8, 8, 7 ]
ages = df['ages']

In [25]:
np.mean(ages)

10.0

In [26]:
ages.var()

6.333333333333333

In [27]:
(ages.var()) **.5

2.516611478423583

In [29]:
2.516611478423583/(7 ** 0.5)

0.9511897312113418

In [32]:
df['ages'] = [14, 12, 11, 10, 8, 7, 1 ]
ages = df['ages']

In [33]:
np.mean(ages)

9.0

In [34]:
ages.var()

18.0

In [35]:
(ages.var()) **.5

4.242640687119285

In [37]:
((ages.var()) **.5) / (7 ** 0.5)

1.6035674514745462

In [38]:
np.median(ages)

10.0

4) This change in actors would not change my centeral tendency, I would still stick with the median because it would remove the largest outliers.  However, I would change my varience to Standard Error, because it would show that the new number '1' is skewing the data and is far from the other data points

5) First, I dont think there is a very good answer to this question, because the  samples seem very bais. I dont think it represents the overall population very well.  However, if I had to, I would find the mean between the 4 magz then add/subtract the standard error. numbers below

In [43]:
ef['fans'] = [20, 23 , 17, 5]
fans = ef['fans']

In [48]:
fanmean = np.mean(fans)

In [45]:
fans.var()

62.25

In [46]:
((fans.var()) **.5)

7.88986691902975

In [49]:
fanse = ((fans.var()) **.5)/(4 **0.5)

In [50]:
fanhigh = fanmean + fanse
fanlow = fanmean - fanse

In [52]:
print(fanhigh)
print(fanlow)

20.194933459514875
12.305066540485125


##    Challenge 3:2:4

In [53]:
1/16

0.0625

1) All 3 would have a 1 in 16 chance or 06.25% chance of landing in those particular orders

2) 24/45

3) plane ride | crash 

In [54]:
plane_ride = 1/10

In [59]:
crash = 1/20000

In [63]:
1/20000 *100 

0.005

In [66]:
prob = plane_ride * crash 

In [67]:
prob * 100

0.0005

Bernice has a 1 in 200000 chance of being in a plane crash or .0005% chance 

4) The problem with this information comes from the way the data scientist have gathered their information.  They do not take into account that their survey adds to the time spend on their site.  Also, the people who are willing to fill out the survey are not a total representation of the population. For example anyone in a hurry or quickly visiting the sight will not have time for the survey. Causing their shorter times to be left out of the data set, leading to a bais.

##   Challenge 3:2:6

1) this is a 0.49 % from the a positive positive and a 10% from a positive negative. So, total there\ is a 10.49% of a positive result

2) positive, positive = 0.49%

3) negative, negative = 90%

4) 10% + .01% = 10.01%



In [68]:
(.98 *.005)

0.0049

In [71]:
(1/10)*(.98 *.005)

0.00049

In [72]:
.02 * .005

0.0001

##   Challenge 3:2:7

1) First, for the Amsterdam data source, there are a few flaws I can see right away.  The question, What are popular neighborhoods in Amsterdam? This question seems like it would be better suited for a much longer time period. Instead of just one day dec 24th.  Second, this date itself causes many baises.  Being the day before a massive cultural holiday it will definetly influence the type of people involved in the data set.  For this example I would change the question to "What are the most popular Christmas destinations in Amsterdam" 

2) For the second data set, I think under normal currcumstances, this question would be fine.  However, being the day after a massive 'terrorist attack' on one of the cities participating in the survey. I think this data would be unusable.  But, if the question was change to see how the recent events affected mental health that would be a much better and it would still help the mental health services.|

3) For the last dataset, I think that this question is way to vague. There are endless possible answer to this question. The way this question is structured there are very few measurable variables.  If this survey was multiple choices so you had more quantitative data to work with.  Then you could find out what aspects/features are drawing the largest percent of customers and what type of customers.  As a bar, knowing what features attracted female customers would be very advantagous.

##   Challenge 3:2:8

In [1]:
right = 1/3
wrong = 2/3

right_two = 1/2
wrong_2 = 1/2

For the Monty Hall problem, I guess I dont get what the challenge is suppose to be.  If you make it past the first round with is 33% chance then you have a 50% chance on either door so I dont think it matters with one you choose.  Personally, I would just stick to the same one from the start.